controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -761,6 +761,12 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 		// No need to create PreprovisioningImage if host is not yet registered
 		// or is externally provisioned
 		preprovImgFormats = nil
+	case metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
```

Comment:
I suspect we should add Deleting to the list of states on l. 760 where we unconditionally avoid creating a PreprovisioningImage.